### PR TITLE
Added powershell here strings to powershell-restmethod codegen

### LINKED ISSUE #322: [PowerShell] Simplify generated code by using here-strings
The current implementation of the PowerShell generator requires a lot of escapes ` ``" `  for each quote and linefeed
this makes the generated core hard to read and change and verify.

**Describe the solution you'd like**
get rid of the escapes by using a [PowerShell here-strings](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-5.1#here-strings) ` @" .... "@`
I would preferer the _double-quoted_ version as that will support variable replacement.

**Proposed use of here-string:**
``` PowerShell
$body = @"
{
       "type":"message",
       "attachments":[
          {
             "contentType":"application/vnd.microsoft.card.adaptive",
             "contentUrl":null,
             "content":
    // insert card here          
    {
      "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
      "type": "AdaptiveCard",
      "version": "1.0",
      "body": [
        {
          "type": "TextBlock",
          "text": "Publish Adaptive Card schema",
          "weight": "bolder",
          "size": "medium"
        }
    ]
    //up to here 
          }
       ]
    }
"@
```

**Current use of many escapes:** 
``` PowerShell
$body = "{
`n   `"type`":`"message`",
`n   `"attachments`":[
`n      {
`n         `"contentType`":`"application/vnd.microsoft.card.adaptive`",
`n         `"contentUrl`":null,
`n         `"content`":
`n// insert card here          
`n{
`n  `"$schema`": `"http://adaptivecards.io/schemas/adaptive-card.json`",
`n  `"type`": `"AdaptiveCard`",
`n  `"version`": `"1.0`",
`n  `"body`": [
`n    {
`n      `"type`": `"TextBlock`",
`n      `"text`": `"Publish Adaptive Card schema`",
`n      `"weight`": `"bolder`",
`n      `"size`": `"medium`"
`n    }
`n]
`n
`n//up to here 
`n
`n      }
`n   ]
`n}"
```

### PR DESCRIPTION
**Pull request by karmanya007**
Fixes #322 .
Got rid of the escapes by using a [PowerShell here-strings](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-5.1#here-strings) in powershell-restmethod codegen.

Request 1
```
method: 'POST',
    header: [
        {
            key: 'Content-Type',
            value: 'text/plain' 
        }
    ],
    url:{
        raw: 'https://mockbin.org/request',
        protocol: 'https',
        host: [
            'mockbin',
            'org'
        ],
        path: [
            'request'
        ]
    },
    body: {
        'mode': 'raw',
        'raw': 'Hello world'
    },
    description: 'Description'
```
Conversion
```sh
$headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
$headers.Add("Content-Type", "text/plain")

$body = @"Hello world"@

$response = Invoke-RestMethod 'https://mockbin.org/request' -Method 'POST' -Headers $headers -Body $body
$response | ConvertTo-Json
```

Request 2
```
method: 'POST',
    header: [],
    body: {
        mode: 'graphql',
        graphql: {
            query: '{ body { graphql } }',
            variables: '{"variable_key": "variable_value"}'
          }
        },
    url: {
        raw: 'http://postman-echo.com/post',
        protocol: 'http',
        host: [
            'postman-echo',
            'com'
        ],
        path: [
            'post'
        ]
    }
```
Conversion
```sh
$headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
$headers.Add("Content-Type", "application/json")

$body = @"{"query":"{ body { graphql } }","variables":{"variable_key":"variable_value"}}"@

$response = Invoke-RestMethod 'http://postman-echo.com/post' -Method 'POST' -Headers $headers -Body $body
$response | ConvertTo-Json
```

### CODE REVIEW

File: codegens/js-xhr/npm-shrinkwrap.json [hunk 1]
```diff
@@ -4,6 +4,53 @@
   "lockfileVersion": 1,
```
Comment (umeshp7) [inline codegens/js-xhr/npm-shrinkwrap.json:4]: Can we remove these changes from this PR?

File: codegens/nodejs-native/npm-shrinkwrap.json [hunk 2]
```diff
@@ -4,28 +4,10 @@
   "lockfileVersion": 1,
```
Comment (umeshp7) [inline codegens/nodejs-native/npm-shrinkwrap.json:4]: Can we remove these changes from this PR?

File: test/codegen/newman/fixtures/formdataFileCollection.json [hunk 3]
```diff
@@ -1,44 +1,47 @@
...
+          "mode": "formdata",
+          "formdata": [
+            {
               "key": "single file",
               "value": "",
               "type": "file",
-              "src": "<file path>"
-						},
-						{
+              "src": "/mnt/c/Users/INVICTUS/Desktop/GSoC/postman-code-generators/dummyFile1.txt"
```
Comment (umeshp7) [inline test/codegen/newman/fixtures/formdataFileCollection.json:20]: this should be `<file path>` There is a script which takes care of adding the path from the CI here.

File: test/codegen/newman/fixtures/formdataFileCollection.json [hunk 4]
```diff
@@ -52,7 +55,7 @@
               "key": "binary file",
               "value": "",
               "type": "file",
-              "src": "<file path>"
+              "src": "/mnt/c/Users/INVICTUS/Desktop/GSoC/postman-code-generators/dummyBinaryFile"
```
Comment (umeshp7) [inline test/codegen/newman/fixtures/formdataFileCollection.json:58]: this should be `<file path>` There is a script which takes care of adding the path from the CI here.

**Review by umeshp7 — CHANGES_REQUESTED** [review]
@karmanya007 Can you add a sample output snippet with the new changes?
Also could you add/update unit tests?

**Review by karmanya007 — COMMENTED** [review]
@umeshp7 Changes added.

**Comment by karmanya007** [comment]
@umeshp7 All checks have passed and I have added the sample outputs.

**Review by Toronto00 — APPROVED** [review]

**Comment by webholik** [comment]
@karmanya007 I have enabled newman tests for poweshell codegen. These tests run the generated snippet in powershell and compare the output with that generated by newman. You can see in the [CI logs](https://travis-ci.com/github/postmanlabs/postman-code-generators/builds/194594824) that some of the tests are failing due to incorrect escaping of characters in the string. We have to get those tests to pass before we can merge your changes.

**Comment by karmanya007** [comment]
> @karmanya007 I have enabled newman tests for poweshell codegen. These tests run the generated snippet in powershell and compare the output with that generated by newman. You can see in the [CI logs](https://travis-ci.com/github/postmanlabs/postman-code-generators/builds/194594824) that some of the tests are failing due to incorrect escaping of characters in the string. We have to get those tests to pass before we can merge your changes.

@webholik On it!

**Comment by umeshp7** [comment]
@karmanya007 Any updates here? Looks like there are some conflicts.

**Comment by karmanya007** [comment]
@umeshp7 Had a marriage in my family so I was not active as much. I know the fix. Just have to implement it. Will do so shortly.

**Comment by karmanya007** [comment]
Getting these three errors. @Toronto00 , @umeshp7 Could you guide me in the correct direction on how to approach these conflicts.

**Comment by karmanya007** [comment]
@webholik Are the 1stand 3rd error from wrong newman tests, maybe?

**Comment by webholik** [comment]
@karmanya007 1st error is because you are not escaping `$` character in the output string. If `$` is present in a string then Powershell tries to replace it with a variable. In this case it is replacing it with the value of `$test`, which isn't defined, so it is just putting empty string here.

2nd error again is just wrong escaping. `` ` `` character must be properly escaped.

3rd has been fixed in #421. You should [rebase](https://docs.github.com/en/free-pro-team@latest/github/using-git/about-git-rebase) your PR on top of it.

**Comment by dhwaneetbhatt** [comment]
@karmanya007 We recently pushed changes to use here-strings in Powershell. Could you check if this PR would still be required?

**Pull request closed without merging**